src/core/src/pass/serialize.cpp

```diff
@@ -1076,47 +1077,60 @@ void ngfunction_2_ir(pugi::xml_node& netXml,
             }
         }
         // <layers/output>
-        if ((node->get_output_size() > 0) && !ov::op::util::is_output(node)) {
-            pugi::xml_node output = layer.append_child("output");
-            for (auto& o : node->outputs()) {
-                pugi::xml_node port = output.append_child("port");
-                port.append_attribute("id").set_value(port_id++);
-
-                const auto& rt_info = o.get_tensor().get_rt_info();
-                auto port_element_type =
-                    is_fp16_compression_postponed(rt_info) ? ov::element::f16 : o.get_element_type();
+        if (node->get_output_size() > 0) {
+            auto serialize_tensor_names = [](const std::unordered_set<std::string>& names) -> std::string {
+                auto sorted_names = std::vector<std::string>(names.begin(), names.end());
+                std::sort(sorted_names.begin(), sorted_names.end());
+
+                std::string serialized_names;
+                for (const auto& name : sorted_names) {
+                    if (!serialized_names.empty())
+                        serialized_names += ",";
+                    serialized_names += escape_delim(name);
+                }
+                return serialized_names;
+            };
+
+            if (ov::op::util::is_output(node)) {
+                if (version > 10 && !deterministic) {
+                    // Not serialize output names for deterministic mode (hash) computation as it is optional
+                    // attribute for v11 and not affect on model structure or how it works
+                    if (const auto& names = ov::descriptor::get_assigned_names(node->get_output_tensor(0));
+                        !names.empty()) {
+                        layer.append_attribute("output_names").set_value(serialize_tensor_names(names).c_str());
```

Comment:
Do we need to describe "output_names" property in docs?

Reply:
I think is not required as it is optional and OV handle it automatically.